src/test/test_utils.rs

```diff
@@ -5,6 +5,133 @@
 
 //! Utilities helpful for writing Rust integration tests.
 
+use std::error;
+use std::fmt;
+
+#[derive(Debug)]
```

Comment:
Is it worth having a custom error type here, rather than using `Box<dyn Error>` and letting the test functions return whatever they want? (e.g. see proc_maps.rs for example)

Reply:
I chose a custom error type rather than a `dyn Error` since there's no nice way to pass custom information in a trait object if we want that in the future. I have changed it to use generics instead of a custom type or trait object, so it's more flexible than before, but also allows tests to pass custom information in errors since it uses a concrete type.